- create in the table database to which store the movie objects

visoth99 opened the pull request:
-search through the movie objects by name
-"problem": the duplicate movies in the output because it has difference values of the awards
- the output is the Json format

visoth99 committed:
- create in the table database to which store the movie objects
-search through the movie objects by name
-"problem": the duplicate movies in the output because it has difference values of the awards
- the output is the Json format

vgalbraith commented:
Correct me if I'm wrong, data() looks like the code to search the database, but where is the code that initializes the database with FetchFromCSV?

zedchance commented on src/main/java/api/data.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.Scanner;
+
+//connect the SQL database file with java
+public class data {
```
I'd recommend that you capitalize the name of your class to `Data`, or possibly a name that describes it reads from the database?

vgalbraith replied:
Either `FetchFromDB` or `FetchFromDatabase` would match our current naming pattern.

zedchance commented:
So this is storing the data as JSON in the database?

zedchance commented on oscars.iml:
```diff
@@ -71,6 +73,5 @@
     <orderEntry type="library" scope="TEST" name="Maven: com.vaadin.external.google:android-json:0.0.20131108.vaadin1" level="project" />
     <orderEntry type="library" scope="TEST" name="Maven: org.springframework:spring-test:5.2.10.RELEASE" level="project" />
     <orderEntry type="library" scope="TEST" name="Maven: org.xmlunit:xmlunit-core:2.7.0" level="project" />
-    <orderEntry type="library" name="gson-2.8.5" level="project" />
```
Also it seems that your .iml file removed the gson library?

zedchance replied:
Nevermind, I see it at the top of the file now. It says there are conflicts on this file so I'm wondering if that is it?

visoth99 committed:
-change from the name: "data" to "Data"

visoth99 committed:
-"Fixed the duplicate Value" (which added the award objects together, such as category, name, and winner)

visoth99 committed:
-changed the printValue method to return the ArrayList which added the output from the database to the Arraylist elements because some movies have the same words like Sunrise,Sunrise at Campobello and Sunrise over Tiananmen Square.

vgalbraith commented:
This is working well on my end, you should upload the initialization code though before we merge.

visoth99 committed:
-submit the python files of the creation database
-"location": in the group_project directory

visoth99 committed:
-"delete": the csv file, and text file

visoth99 committed:
-"fixed ,Data class": empty string search which gave an error output.
-"created" : MovieArray class for the movie title list.
-"create" : UpdateDataBase class for update a jSON string.

vgalbraith commented:
Since the rest of the project is utilizing movie objects, I think it's important that the classes dealing with the database do as well.

For the Data() class, which fetches the movies from the database, it should read them from the database into an array list of movie objects, and return them as such, just as FetchFromCSV() returns them. That way the rest of our methods can be used on the output.

For the UpdateDataBase() class, it should be able to take in a movie object, and update the database with any additional information that has been called from OMDb/IMDb (for instance, this can be done by checking if the 'website' field is no longer empty in the movie object). This may also require some kind of initialization of these empty fields in the database that are to be filled with OMDb/IMDb information (you can see which fields a movie object has in the Movie class).

I'm not sure what the MovieArray class is intended to be used for.

vgalbraith commented:
The team agreed that integrating the database at this point will be too ambitious.
